src/name_bazaar/ui/pages/manage_names_page.cljs

```diff
@@ -0,0 +1,256 @@
+    [name-bazaar.ui.components.offering.offering-type-select :refer [offering-type-select]]
+    [name-bazaar.ui.constants :as constants]
+    [name-bazaar.ui.utils :refer [reverse-record-node namehash sha3 normalize strip-root-registrar-suffix valid-ens-name? path-for]]
+    [re-frame.core :refer [subscribe dispatch]]
+    [reagent.core :as r]
+    [cljs-web3.core :as web3]
+    [soda-ash.core :as ui]))
+
+(def ownership-status->text
+  {:ens.ownership-status/empty-name ""
+   :ens.ownership-status/loading "Checking ownership..."
+   :ens.ownership-status/not-ens-record-owner "You are not owner of this name"
+   :ens.ownership-status/not-deed-owner "You don't own this name's locked value"
+   :ens.ownership-status/owner "You are owner of this name"})
+
+(defn default-point-name-form-data [addr]
+  {:ens.record/address (or addr "0x")
+   :ens.record/name ""}
+  )
+
+(defn load-name-ownership [value]
+  (let [full-name (str value constants/registrar-root)
+        node (namehash full-name)]
+    (dispatch [:ens.records/load [node] {:load-resolver? true}])
+    (when (top-level-name? full-name)
+      (dispatch [:registrar.entries/load [(sha3 value)]]))))
+
+(defn load-resolver [address]
+  (let [node (reverse-record-node address)]
+    (dispatch [:ens.records.resolver/load [node]])))
+
+(defn ens-name-text-field [{:keys [:value]}]
+  (fn [{:keys [:on-change :value] :as props}]
+    (let [full-name (when (seq value)
+                      (str value constants/registrar-root))
+          ownership-status @(subscribe [:ens.record/ownership-status full-name])
+          standard-resolver? @(subscribe [:ens.record/standard-resolver? (namehash full-name)])
+          error? (contains? #{:ens.ownership-status/not-ens-record-owner
+                              :ens.ownership-status/not-deed-owner}
+                            ownership-status)]
+      [:div.input-state-label
+       {:class (cond
+                 error? :error
+                 (contains? #{:ens.ownership-status/owner} ownership-status) :success)}
+       ;;[:div (str "resolver>" standard-resolver? "<")]
+       [ens-name-input
+        (r/merge-props
+          {:label "Name"
+           :fluid true
+           :value value
+           :error error?
+           :on-change (fn [e data]
+                        (let [value (aget data "value")]
+                          (when (valid-ens-name? value)
+                            (let [value (normalize value)]
+                              (aset data "value" value)
+                              (on-change e data)
+                              (load-name-ownership value)))))}
+          (dissoc props :ownership-status :on-change))]
+       [:div.ui.label (ownership-status->text ownership-status)]])))
+
+(defn address-text-field [{:keys [:value :load-resolver?]}]
+  (fn [{:keys [:on-change :value] :as props}]
+    (let [is-empty? (or
+                     (empty? value)
+                     (empty-address? value))
```

Comment:
You can add into `empty-address?` checking for empty string. Don't have to use `or` here then